File: benchmarks/pandas/bench_str_encode.py
"""Benchmark: str_encode — str.encode byte-length encoding on 100k strings"""
import json
import time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = [f"hello world {i}" for i in range(ROWS)]
s = pd.Series(data)

for _ in range(WARMUP):
    s.str.encode("utf-8")

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.str.encode("utf-8")
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "str_encode",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_str_encode.ts
/**
 * Benchmark: str_encode — str.encode byte-length encoding on 100k strings
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Array.from({ length: ROWS }, (_, i) => `hello world ${i}`);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  s.str.encode();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.str.encode();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "str_encode",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
